src/platform_impl/windows/event_loop/runner.rs

```diff
@@ -1,51 +1,50 @@
-use std::{any::Any, cell::RefCell, collections::VecDeque, mem, panic, ptr, rc::Rc, time::Instant};
+use std::{
+    any::Any,
+    cell::{Cell, RefCell},
+    collections::{HashSet, VecDeque},
+    mem, panic, ptr,
+    rc::Rc,
+    time::Instant,
+};
 
-use winapi::{shared::windef::HWND, um::winuser};
+use winapi::{
+    shared::{minwindef::DWORD, windef::HWND},
+    um::winuser,
+};
 
 use crate::{
     dpi::PhysicalSize,
     event::{Event, StartCause, WindowEvent},
     event_loop::ControlFlow,
-    platform_impl::platform::{event_loop::EventLoop, util},
+    platform_impl::platform::util,
     window::WindowId,
 };
 
 pub(crate) type EventLoopRunnerShared<T> = Rc<ELRShared<T>>;
 pub(crate) struct ELRShared<T: 'static> {
-    runner: RefCell<Option<EventLoopRunner<T>>>,
-    buffer: RefCell<VecDeque<BufferedEvent<T>>>,
-    redraw_buffer: Rc<RefCell<VecDeque<WindowId>>>,
-}
-struct EventLoopRunner<T: 'static> {
-    control_flow: ControlFlow,
-    runner_state: RunnerState,
-    modal_redraw_window: HWND,
-    in_modal_loop: bool,
-    event_handler: Box<dyn FnMut(Event<'_, T>, &mut ControlFlow)>,
-    panic_error: Option<PanicError>,
-    redraw_buffer: Rc<RefCell<VecDeque<WindowId>>>,
+    thread_msg_target: HWND,
+    wait_thread_id: DWORD,
+    processing_events: Cell<ProcessingEvents>,
+    panic_error: Cell<Option<PanicError>>,
+    control_flow: Cell<ControlFlow>,
+    last_events_cleared: Cell<Instant>,
+    event_handler: Cell<Option<Box<dyn FnMut(Event<'_, T>, &mut ControlFlow)>>>,
+    event_buffer: RefCell<VecDeque<BufferedEvent<T>>>,
+    owned_windows: Cell<HashSet<HWND>>,
 }
 pub type PanicError = Box<dyn Any + Send + 'static>;
 
-pub enum BufferedEvent<T: 'static> {
-    Event(Event<'static, T>),
-    ScaleFactorChanged(WindowId, f64, PhysicalSize<u32>),
-}
-
-#[must_use]
-#[derive(Debug, Clone, Copy)]
-pub enum AreEventsBuffered {
-    EventsBuffered,
-    ReadyToSleep,
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+enum ProcessingEvents {
```

Comment:
the enum seems to duplicate the runner states and the name does not "speak" much an

Reply:
That's a good point - honestly, I'm not entirely sure why I renamed these fields 😅. I've gone ahead and reverted the names to `RunnerState`'s names.